cli/loader_lib.c

```diff
@@ -55,19 +58,21 @@ static void * load_library(const char * rel_path, const char * src_dir, int err)
     strncat(path, rel_path, sizeof(path) - 1);
 
 #if defined(_OS_WINDOWS_)
+#define PATH_EXISTS() !_waccess(wpath, 4)
```

Comment:
We could use [_waccess_s](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/access-s-waccess-s?view=msvc-170) here, to check both whether the file exists (which is all that `_waccess(wpath, 4)` checks) and whether the current user has permission to read it

But I am inclined to say we should just check existence (F_OK), so that any permission errors in opening the file will get reported by `dlerror` as fatal errors for the process, as a more conservative (more noisy) default. WDYT?

Reply:
It looks like the fact that `_waccess_s` exists implies that `_waccess` is deprecated. It's unclear to me how the two are different, other than that the CRT is expected to ensure that the arguments and memory are valid.

On the other point, I agree. Under the assumption that everything is installed correctly, the only way that the lib could become unreadable is if the user did something dumb. It would be better if it didn't silently break in that case.

Reply:
`_waccess` probably only checks for existence and the value of the advisory read-only flag, while `_waccess_s` checks instead the actual access permissions of the file (probably calling https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-accesscheck)

Reply:
ah, shoot, this is loader_lib.c, so we don't have access to msvc here yet anyways, so that point is rather moot anyways. We need to do this the extra-hard way apparently, only using KERNEL32 to write our own `access` function:
```
$ objdump -x libjulia-internal.dll | grep 'DLL Name'
        DLL Name: libjulia.dll
        DLL Name: ADVAPI32.dll
        DLL Name: dbghelp.dll
        DLL Name: IPHLPAPI.DLL
        DLL Name: KERNEL32.dll
        DLL Name: msvcrt.dll
        DLL Name: ole32.dll
        DLL Name: PSAPI.DLL
        DLL Name: libwinpthread-1.dll
        DLL Name: Secur32.dll
        DLL Name: SHELL32.dll
        DLL Name: USER32.dll
        DLL Name: USERENV.dll
        DLL Name: WINMM.DLL
        DLL Name: WS2_32.dll
        DLL Name: libgcc_s_sjlj-1.dll
        DLL Name: libstdc++-6.dll
$ objdump -x libjulia.dll | grep 'DLL Name'
        DLL Name: KERNEL32.dll
```